run/gem5art/run.py

```diff
@@ -315,8 +315,8 @@ def checkArtifacts(self, cwd: str) -> bool:
                 new = artifact.artifact.getHash(cwd / v.path)
                 old = v.hash
 
-            if new != v.hash:
```

Comment:
Is this really a bug? Because, the hash parameter of git artifact is set to git hash as well while registering the artifact as here: https://github.com/darchr/gem5art/blob/5a3a564fce3935052a0d1e4cc4ba2d46fdcaf587/artifact/gem5art/artifact/artifact.py#L158

Reply:
However, this change still makes sense to me from the perspective of code readability

Reply:
Wow, good catch! That makes more sense why it was working before.

I guess we don't need the if/else statement, is what you're saying, right? The missing `self` and the fact that `old` was an unused variable were definitely bugs though :)

Reply:
Yeah, the if/else is unnecessary here.

Reply:
Actually, I think it's still necessary. While we set the hash correctly in an artifact, we don't have that logic in `getHash()` https://github.com/darchr/gem5art/blob/5a3a564fce3935052a0d1e4cc4ba2d46fdcaf587/artifact/gem5art/artifact/artifact.py#L42

So, while we don't need the `old` variable, and we could just use `v.hash`, we *do* need the `if` statement to select the right `new` hash.

Reply:
That makes sense :)